public/app/features/explore/ExploreQueryInspector.tsx

```diff
@@ -71,7 +74,11 @@ export function ExploreQueryInspector(props: Props) {
     value: 'query',
     icon: 'info-circle',
     content: (
-      <QueryInspector data={queryResponse} onRefreshQuery={() => props.runQueries({ exploreId: props.exploreId })} />
+      <QueryInspector
+        instanceId={isMixed ? mixedRequestId(0, requestIdGenerator(exploreId)) : requestIdGenerator(exploreId)}
```

Comment:
I think that ideally `ExploreQueryInspector` should not depend on the implementation details of the data source (and check if it's mixed or not). What is more, it seems that any data source can override the request ID, not only mixed so we cannot guarantee it'd work in every case.

Would it be possible to instead of passing the generated `requestId`, keep track of the returned, latest response's `requestId` after the query is run and pass only the requestId to `ExploreQueryInspector`?

Reply:
🤦 the query inspector has the response and thus the requestId already. cleaned this up quite a bit

Reply:
@ifrost actually, we can't do this. The mixed datasource takes whatever the incoming request ID is and adds a prefix to it. So whatever requestID explore makes and sends to the query and the query inspector gets changed by the datasource. Thus, the response request ID is not the same - it always is the requestID post that change.

Everything has a requestID and I can ensure the logic the mixed datasource uses and what gets passed around is the same, but I can't rely on requestId alone for this. Let me know if you have any other ideas - I know ideally the ExploreQueryInspector shouldn't know/care if the datasource is mixed but Id rather have this logic kept within Explore.

Reply:
> The mixed datasource takes whatever the incoming request ID is and adds a prefix to it.

I meant changing the logic so instead of passing `requestId` to the inspector that "should match" the one that was used: make a request and take the `requestId` but from the response object after the request is made (maybe [this](https://github.com/grafana/grafana/blob/6ee2bd4d29413882164db55833034985ad30b4e6/packages/grafana-data/src/types/panel.ts#L54)?) and store in redux store. 

This, however, assumes we can get the same response object with the same `requestId` in 2 places:
- as result of`runRequest`,
- and in the inspector stream. 

But if I understand correctly, you're saying there's no way to match responses like this.

I think QueryInspector should not parse `response.config.url`, and rather that information should be attached to the response. We should be able to get the same information attached to responses we handle in Explore. In other words I think to solve this problem we need to have a clear way to have an `id` (like `requestId`) attached to the response to can clearly identify it in various places. This way Explore would know what was the latest loaded `id` in a pane (by storing it in the state) and pass the same id to the inspector. But it seems that would require a lot of refactoring out of the scope.